### DISCUSSION TITLE
Launching via MvcBoilerPlate profile throws BaddHttpRequestException

### DISCUSSION THREAD

**Discussion by ketrex2** [Dotnet-Boxed/Templates#125] [Q&A]
Website works fine when using iisExpress launch profile. When I try to launch the website via the MvcBoilerPlate launch profile, I get the following output in the console and the website will not load:

```
info: Microsoft.AspNetCore.Server.Kestrel[17]
      Connection id "0HKUITSU018MS" bad request data: "The input string contains non-ASCII or null characters."
Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException: The input string contains non-ASCII or null characters.
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure.MemoryPoolIteratorExtensions.GetAsciiString(MemoryPoolIterator start, MemoryPoolIterator end)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame.TakeStartLine(SocketInput input)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame`1.<RequestProcessingAsync>d__2.MoveNext()
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Connection processing ended abnormally
Microsoft.AspNetCore.Server.Kestrel.BadHttpRequestException: The input string contains non-ASCII or null characters.
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure.MemoryPoolIteratorExtensions.GetAsciiString(MemoryPoolIterator start, MemoryPoolIterator end)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame.TakeStartLine(SocketInput input)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame`1.<RequestProcessingAsync>d__2.MoveNext()
```

MVC 6 running netcoreapp1.0
Visual Studio 2015 Version 14.0.25425.01 Update 3 
dotnet version 1.0.0-preview2-003121

I am using the default values and have made no modifications to the project prior to launching. This error is happening on first build after creating the project.

**Comment by RehanSaeed** [comment]
Do either of these help?
- [error on connection in kestrel - non-ascii or null character](https://github.com/aspnet/KestrelHttpServer/issues/949).
- [Running Kestrel Using HTTPS](https://github.com/aspnet/KestrelHttpServer/issues/969). This is unlikely since you say you made no changes.

**Comment by ketrex2** [comment]
Whoops! Yes, I forgot to implement my self-signed cert.

I see you have instructions in the ReadMe.html for generating a cert, which is great, but I wonder if it would help users to have a reminder that Kestrel requires the manual invocation of a cert in the WebHostBuilder?

I'd be happy to do a pull if you're busy but think this is worthwhile. This is a fantastic project.

**Comment by RehanSaeed** [comment]
My plan is to add a 'Kestrel over HTTPS' feature which adds the Microsoft.AspNetCore.Server.Kestrel.Https NuGet package and the code in Program.cs to setup a certificate. The only part missing, as you say, will be the certificate itself. Not sure what to do there, probably I will throw a NotImplementedException explaining what to do. Adding a note in the ReadMe is unlikely to get read on first run of the project.

Any PR's are always welcome.

**Comment by RehanSaeed** [comment]
Updated the [TODO](https://github.com/RehanSaeed/ASP.NET-MVC-Boilerplate/blob/master/TODO.md) with the above feature.

**Comment by ketrex2** [comment]
That's great.
I just exported the cert generated by IIS Express since it's already assigned to localhost and it's in the certificate store by default, so it just works and I don't have to install any self-signed certs of my own. Instructions on how to export and import the existing IIS Express cert may be the best way to go. Just my two cents.

**Comment by RehanSaeed** [comment]
Glad you got it working!
